File: backend/models/Museum.js
import mongoose from "mongoose";

const museumSchema = new mongoose.Schema({
  museumName: String,

  email: {
    type: String,
    required: true,
    unique: true,
    lowercase: true
  },

  phone: String,
  location: String,

  password: String,

  isVerified: {
    type: Boolean,
    default: false
  },

  otp: String,
  otpExpiry: Date,

  resetOTP: String,
  resetOTPExpiry: Date
});

export default mongoose.model("Museum", museumSchema);

File: backend/app/db/models/museum.py
from pydantic import BaseModel, EmailStr, Field
from typing import Optional
from datetime import datetime

class MuseumDocument(BaseModel):
    museumName: str
    email: EmailStr
    phone: str
    location: str
    password: Optional[str] = None
    is_verified: bool = False
    otp: Optional[str] = None
    otp_expiry: Optional[datetime] = None
    reset_otp: Optional[str] = None
    reset_otp_expiry: Optional[datetime] = None
    reset_verified: bool = False
    created_at: datetime = Field(default_factory=datetime.utcnow)